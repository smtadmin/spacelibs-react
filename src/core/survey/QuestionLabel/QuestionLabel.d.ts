export default QuestionLabel;
declare class QuestionLabel {
    constructor(props: any);
    render(): JSX.Element;
}
declare namespace QuestionLabel {
    namespace defaultProps {
        const isRequired: boolean;
    }
    namespace propTypes {
        export const number: any;
        export const label: any;
        export const helperText: any;
        const isRequired_1: any;
        export { isRequired_1 as isRequired };
    }
}
