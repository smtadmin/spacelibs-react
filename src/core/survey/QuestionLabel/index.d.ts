export default QuestionLabel;
import QuestionLabel from "./QuestionLabel";
