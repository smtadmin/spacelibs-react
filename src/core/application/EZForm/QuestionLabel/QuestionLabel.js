/*
 * File: /src/core/survey/QuestionLabel/QuestionLabel.js
 * Version: 0.0.2
 * Project: @siliconmtn/spacelibs-react
 * Description: Question Label component
 * File Created: Wednesday, 10th February 2021 11:08 am
 * Author: tyler Gaffaney (tyler.gaffaney@siliconmtn.com)
 * -----
 * Last Modified: Friday, 28th May 2021 11:56 am
 * Modified By: tyler Gaffaney (tyler.gaffaney@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */

import React from 'react';
import PropTypes from 'prop-types';
import Tooltip from '../../../input/Tooltip';

/**
 * QuestionLabel Component
 * 
 * @param {*} props - Component props
 * @returns {*} Component
 */
function QuestionLabel(props) {
	/**
	 * Returns the number label
	 *
	 * @returns {*} Number String
	 */
	function getNumber(){
		if(props.number == null)
			return null;
		else 
			return props.number + ".";
	}

	return <div 
	className={"question-label"}>
		{getNumber()} {props.label} {props.required && <span style={{color:'red'}}>*</span>} <Tooltip text={props.helperText}/>
	</div>;
}

QuestionLabel.defaultProps = {
	required: false
};

QuestionLabel.propTypes = {
	number: PropTypes.number,
	label: PropTypes.string.required,
	helperText: PropTypes.string,
	required: PropTypes.bool
};

export default QuestionLabel;