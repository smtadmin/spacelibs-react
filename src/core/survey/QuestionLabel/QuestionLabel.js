/*
 * File: /src/core/survey/QuestionLabel/QuestionLabel.js
 * Version: 0.0.2
 * Project: @siliconmtn/spacelibs-react
 * Description: INSERT DESCRIPTION
 * File Created: Wednesday, 10th February 2021 11:08 am
 * Author: tyler Gaffaney (tyler.gaffaney@siliconmtn.com)
 * -----
 * Last Modified: Wednesday, 10th February 2021 11:12 am
 * Modified By: tyler Gaffaney (tyler.gaffaney@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */

import React from 'react';
import PropTypes from 'prop-types';
import Tooltip from '../../input/Tooltip';

/**
 * Label component for survey questions
 */
class QuestionLabel extends React.Component {

	constructor(props){
		super(props);
	}

	render(){
		return <div style={{'fontFamily':'arial'}}>
			{this.props.number}. {this.props.label} {this.props.isRequired && <span style={{color:'red'}}>*</span>} <Tooltip text={this.props.helperText}/>
		</div>;
	}
}

QuestionLabel.defaultProps = {
	isRequired: false
};

QuestionLabel.propTypes = {
	number: PropTypes.number.isRequired,
	label: PropTypes.string.isRequired,
	helperText: PropTypes.string,
	isRequired: PropTypes.bool
};

export default QuestionLabel;