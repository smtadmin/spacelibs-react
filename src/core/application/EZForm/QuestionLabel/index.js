/*
 * File: /src/core/survey/QuestionLabel/index.js
 * Version: 0.0.2
 * Project: @siliconmtn/spacelibs-react
 * Description: Index files are used to make exporting easier
 * File Created: Wednesday, 10th February 2021 11:08 am
 * Author: tyler Gaffaney (tyler.gaffaney@siliconmtn.com)
 * -----
 * Last Modified: Friday, 26th February 2021 2:09 pm
 * Modified By: tyler Gaffaney (tyler.gaffaney@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */

import QuestionLabel from './QuestionLabel';

export default QuestionLabel;