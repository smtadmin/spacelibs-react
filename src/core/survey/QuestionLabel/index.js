/*
 * File: /src/core/survey/QuestionLabel/index.js
 * Version: 0.0.2
 * Project: @siliconmtn/spacelibs-react
 * Description: INSERT DESCRIPTION
 * File Created: Wednesday, 10th February 2021 11:08 am
 * Author: tyler Gaffaney (tyler.gaffaney@siliconmtn.com)
 * -----
 * Last Modified: Wednesday, 10th February 2021 11:12 am
 * Modified By: tyler Gaffaney (tyler.gaffaney@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */

import QuestionLabel from './QuestionLabel';

export default QuestionLabel;