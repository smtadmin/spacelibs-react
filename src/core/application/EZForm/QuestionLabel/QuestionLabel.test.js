/*
 * File: /src/core/application/EZForm/QuestionLabel/QuestionLabel.test.js
 * Version: 0.0.2
 * Project: @siliconmtn/spacelibs-react
 * Description: INSERT DESCRIPTION
 * File Created: Monday, 1st March 2021 8:50 am
 * Author: tyler Gaffaney (tyler.gaffaney@siliconmtn.com)
 * -----
 * Last Modified: Monday, 1st March 2021 9:17 am
 * Modified By: tyler Gaffaney (tyler.gaffaney@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */

import QuestionLabel from "./QuestionLabel";
import React from 'react';
import { render } from '@testing-library/react';
import "@testing-library/jest-dom";

/**
 * Checks that the button renders and the current snapshot matches the previous one.
 */
it("Renders with no props", ()=> {
	const { baseElement, getByText } = render(<QuestionLabel number={1} label={"What is your name?"} isRequired={false}/>);
	expect(baseElement).toBeTruthy();
	expect(baseElement).toMatchSnapshot();
	expect(getByText("1. What is your name?")).toBeInTheDocument();
});

it("Renders with required prop", () => {
    const { baseElement, getByText } = render(
        <QuestionLabel
            number={1}
            label={"What is your name?"}
            isRequired={true}
        />
    );
    expect(baseElement).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
    expect(getByText("*")).toBeInTheDocument();
});