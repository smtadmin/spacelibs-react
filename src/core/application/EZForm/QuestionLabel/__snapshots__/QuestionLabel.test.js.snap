// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with no props 1`] = `
<body>
  <div>
    <div
      class="question-label"
    >
      1.
       
      What is your name?
       
       
    </div>
  </div>
</body>
`;

exports[`Renders with required prop 1`] = `
<body>
  <div>
    <div
      class="question-label"
    >
      1.
       
      What is your name?
       
      <span
        style="color: red;"
      >
        *
      </span>
       
    </div>
  </div>
</body>
`;
